aztec/src/main/kotlin/org/wordpress/aztec/source/SourceViewEditText.kt

```diff
@@ -184,6 +191,12 @@ open class SourceViewEditText : android.support.v7.widget.AppCompatEditText, Tex
     }
 
     fun displayStyledAndFormattedHtml(source: String) {
+        initialContentHolder?.let {
+            if (it.needToSetInitialValue()) {
+                it.setInitialContent(source)
+            }
+        }
+
         val styledHtml = styleHtml(Format.addSourceEditorFormatting(source, isInCalypsoMode))
```

Comment:
I found something interesting here. I think we should first run `Format.addSourceEditorFormatting` and save the output as the `initialContent`, to make sure we account for Calypso mode, something like this (haven't tested):

```
fun displayStyledAndFormattedHtml(source: String) {

        val formattedSource = Format.addSourceEditorFormatting(source, isInCalypsoMode)
        initialContentHolder?.let {
            if (it.needToSetInitialValue()) {
                it.setInitialContent(formattedSource)
            }
        }

        val styledHtml = styleHtml(formattedSource)
[...]
```

I'm saying this because I found an interesting situation testing on WPAndroid with the corresponding PR, as follows.

Write a post with this content on the web:
```
<p>Dzssgshdzx did</p>
<p>Cudjddgn</p>
```

Now on WPAndroid:
1. open the post
2. don't edit anything
3. switch to HTML
4. tap BACK
5. -> observe the `hasChanges` method on the source editor returns `true`

This is due to how we handle Calypso mode, we need to calculate the initial hash taking calypso mode into account.

Please note that, in WPAndroid the actual Post does not get updated to the server, because there's still some code in WPAndroid that prevents this from happening, but formally we should probably make sure that Calypso mode conversions are not confounded by Aztec itself as a "change".

Given the code in WPAndroid saves this from happening, I think we can probably live with this as is, but I thought it's worth mentioning, what do you think?

Reply:
Thanks Mario for the testing effort and for the proposed solution. I tried with it with no luck. since the transformation to Calypso HTML, and Jsoup parsing is changing the content. 
Making  hard to get it back and compare it.